### PR TITLE
Infinite Gun Culling: Gun Kit Edition (#56133) | TESTED WORKING

### PR DESCRIPTION
**Pull request by Simplehorror**
Port of https://github.com/tgstation/tgstation/pull/56133
# Document the changes in your pull request
Every prototype firearm in the game now utilizes crafting to construct it, usually sacrificing a standard version of that firearm in order to produce the new one. The guns are made using gun part kits you print from the security protolathe (or buy, in the case of hellfire lasers).

This makes it harder for cargo to straight up buy weaponry, and nerfs the beam rifles so you cant print 100 and then annihilate a blob (since that's all it's good for)

# Wiki Documentation

Guns can no longer be printed and must be crafted

# Changelog

:cl:  necromanceranne, ported by Simplehorror
add: Guns are no longer directly printable from the protolathe.
add: Gun kits to craft the various prototype firearms once printable from the security protolathe.
tweak: Large energy crossbows now are bulky, like most other printed weapons.
tweak: Crafted weapons now come with a firing pin.
/:cl:

### CODE REVIEW

**Comment by Cartlord** [comment]
Did you miss the part where beam rifles have been nerfed to HELL already?

**Comment by Bibby0110** [comment]
Please add a direct link to the PR in your PR body :)
https://github.com/tgstation/tgstation/pull/56133

**Comment by Simplehorror** [comment]
> Did you miss the part where beam rifles have been nerfed to HELL already?

Did you miss the part where none of the nerfs mattered?

**Comment by N3D6** [comment]
He actually did it

**Comment by Mqiib** [comment]
image


>advaned energy gun

**Comment by ToasterBiome** [comment]
Bad

**Comment by Marmio64** [comment]
Only part I can agree with is making large energy crossbows bulky. Otherwise, this is another unneeded PR.

File: code/datums/components/crafting/recipes.dm [hunk 1]
```diff
@@ -202,6 +202,140 @@
...
+/datum/crafting_recipe/tempgun/New()
+	..()
+	blacklist += subtypesof(/obj/item/gun/energy/e_gun)
+
+/datum/crafting_recipe/beam_rifle
+	name = "Particle Acceleration Rifle"
+	tools = list(TOOL_SCREWDRIVER, TOOL_WIRECUTTER)
+	result = /obj/item/gun/energy/beam_rifle
+	reqs = list(/obj/item/gun/energy/e_gun = 1,
+				/obj/item/assembly/signaler/anomaly = 2,
```
Comment (SomeguyManperson) [inline code/datums/components/crafting/recipes.dm:240]:
```suggestion
```

**Comment by Runian** [comment]
Besides requiring 2 anomaly cores to make a PA rifle, I don't see how this is super terrible. If security wants the better guns, they have to spend a bit more time instead of just printing the gun and the firing pin. I suggest removing anomaly core requirement for PA rifle or port over something that eases access to anomalies (https://github.com/tgstation/tgstation/pull/51432).

**Comment by JamieD1** [comment]
Way too much of a nerf, anomalies are exceedingly rare as it is.

**Comment by Simplehorror** [comment]
> Way too much of a nerf, anomalies are exceedingly rare as it is.

how about you tell me that before closing my PR

**Pull request closed without merging**